### ISSUE TITLE
Reference AP from Vue.js

### ISSUE THREAD

**Issue by chrdunne** [spartez/vue-aui#2]
Hi

I would like to build an Atlassian Connect add-on using Vue.js - I have the basic functionality working. I'd like to use your components, but the thing that is puzzling me is how I reference the AP Javascript library from a Vue.js component e.g. the call the JIRA REST API using AP.request().

I am starting with a simple static add-on as I'm not planning in having a backend. So I am using the connect-loader method provided in the Atlassian tutorials. I am just unclear how I can reference this later, or in any components.

Any help or advice would be greatly appreciated.

Thanks
Chris

**Comment by damienix** [comment]
Hi Chris, nice to see one of our clients here :)

Honestly, I don't have experience with static add-ons. And this library only provides AUI components for your Vue application. This library doesn't provide Vue library itself but requires it as peer dependency. This means that you need to include vue library yourself. This may be via webpack (or other bundler) or by a script tag.

After loading `atlassian-connect/all.js` script you should have access to global AP variable. In regards to the vue file components, do you use webpack to bundle them or html templates? Either way, you should just have an access to global `AP` variable in any scripts executed after that, including Vue constructor's params of your components.

**Comment by chrdunne** [comment]
Thanks a lot for the quick reply.

I understand that I need to install the vue-aui library as a dependency -
that's fine. I'm using the webpack Vue project template.

I thought that I should have access to the global AP variable, but for some
reason I'm getting an undefined reference error.

I'm sure I must be doing something wrong.

Thanks again
Chris

Chris Dunne
Ph: 353-86-8217982
http://ie.linkedin.com/in/chrdunne

**Comment by damienix** [comment]
Please make sure that your `all.js` script is properly downloaded before your Vuejs app starts. I think in your case connect-loader script is that one loading `all.js` so make sure it's defined before you include webpack bundle.

**Comment by chrdunne** [comment]
Thanks Damien.

Yes, you are correct. But my original problem was that I had chosen to use
ESLint when setting up the project template. So the project was failing to
compile because it didn't recognise the AP reference. I am sure there is
some way to configure webpack to get around this but to be honest I don't
really understand webpack yet. So I recreated the project, without ESLint
turned on.

Then once I had this set up, and added the READ scope to the connect
descriptor, I was able to use AP in the App component. I haven't tried it
yet with other components but I assume it will be the ok.

Thanks
Chris

Chris Dunne
Ph: 353-86-8217982
http://ie.linkedin.com/in/chrdunne
